SportsHub/UnitTests/Controllers/CommentControllerTests.cs

```diff
@@ -0,0 +1,101 @@
+﻿using AutoMapper;
+using FluentValidation;
+using FluentValidation.Results;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using SportsHub.Api.Controllers;
+using SportsHub.Api.Validations;
+using SportsHub.AppService.Authentication.Models.DTOs;
+using SportsHub.AppService.Services;
+using SportsHub.Domain.Models;
+using UnitTests.MockData;
+using UnitTests.Utils;
+using Xunit;
+
+namespace UnitTests.Controllers
+{
+    public class CommentControllerTests
+    {
+        private IMapper _mapper;
+        private readonly CommentController _commentController;
+        private readonly Mock<ICommentService> _commentService;
+        private readonly Mock<IValidator<PostCommentDTO>> _commentValidator;
+        private readonly Mock<IGenerateModelStateDictionary> _generateModelStateDictionary;
+
+        public CommentControllerTests()
+        {
+            if (_mapper == null)
+            {
+                var mappingConfig = new MapperConfiguration(x => x.CreateMap<Comment, PostCommentDTO>());
+                IMapper mapper = mappingConfig.CreateMapper();
+                _mapper = mapper;
+            }
+
+            _commentService = new Mock<ICommentService>();
+            _commentValidator = new Mock<IValidator<PostCommentDTO>>();
+            _generateModelStateDictionary = new Mock<IGenerateModelStateDictionary>();
+            _commentController = new CommentController(_commentService.Object, _mapper, _commentValidator.Object, _generateModelStateDictionary.Object);
+        }
+
+        [Fact]
+        public async Task GetByArticleAsync_CommentsForProvidedActicleExist_ReturnsOkStatus()
+        {
+            //Arrange
+            int acticleId = 5;
+            var comments = CommentMockData.GetForArticle();
+            _commentService.Setup(service => service.GetByArticleAsync(acticleId)).ReturnsAsync(comments);
+
+            //Act
+            var result = await _commentController.GetByArticleAsync(acticleId);
+            var resultObject = TestHelper.GetObjectResultContent<IEnumerable<Comment>>(result);
```

Comment:
How this line of code relates to `Act`?

Reply:
It does not, I've moved it further down the file.